### ISSUE TITLE
Envkey community pre-compiled binary cannot run with env set

### ISSUE THREAD

**Issue by trungklam** [envkey/envkey#54]
I'm trying to host envkey community but got stuck by this issue. Following the guide:
-  Start from pre-compiled binary: execute the envkey-api file.
With the following environment variables already set in .env:

```
COMMUNITY_AUTH_HASH= ...
DATABASE_CREDENTIALS_JSON={"user": "user", "password": "pass"}
DATABASE_HOST=example.database.windows.net
DATABASE_NAME=envkey
DATABASE_PORT=1433
SMTP_TRANSPORT_JSON=...
```
I'm using a Azure SQL server, and running the community api in my local to test it first. the DB "envkey" already created in the server. SQL server port is 1433.
here is my Dockerfile
```
# syntax=docker/dockerfile:1.4
FROM ubuntu:22.04
ENV TZ="Asia/Singapore"
WORKDIR /usr/src/app
COPY ./envkey-api .
EXPOSE 3000
EXPOSE 3002
ENTRYPOINT [ "./envkey-api" ]
```
Since I dont see the use of port 3001 so I removed it from the original dockerfile packed with community api.
Using docker run to test:
```docker run -it --env-file=.env envkey:1.0.0```
 
And here is the result
```
❯ docker run -it --env-file=.env envkey:1.0.0
{"msg":"envkey-api binary start","api_binary_version":"2.2.3","built_from":"b1e05d63d59152522f20f353c0f80e2dedda0c89","bundle_type":"api.community","built_at":"2022-12-20T17:06:45Z"}
NODE_ENV environment variable is required

/tmp/webpack:/Users/daneschneider/projects/envkey/v2/public/app/api/shared/src/env.ts:18
      throw new Error(msg);
            ^
Error: NODE_ENV environment variable is required
    at Object.ensureEnv (/tmp/webpack:/Users/daneschneider/projects/envkey/v2/public/app/api/shared/src/env.ts:18:13)
    at Object.call (/tmp/webpack:/src/startup.ts:8:1)
    at a (/tmp/webpack:/webpack/bootstrap:19:22)
    at Object.call (/tmp/webpack:/src/main_api_community.ts:17:1)
    at __webpack_require__ (/tmp/webpack:/webpack/bootstrap:19:22)
    at /tmp/webpack:/webpack/bootstrap:83:10
    at Object.<anonymous> (/tmp/4667234118124935090:2:968)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
Fatal error
exit status 1
```

I've tried to set the NODE_ENV=production and got this error. Seems It cant connect to the DB but I honestly have no idea why. I tested the connection between my local and Azure SQL Server and was successfully connected to envkey DB.
```
{"msg":"envkey-api binary start","api_binary_version":"2.2.3","built_from":"b1e05d63d59152522f20f353c0f80e2dedda0c89","bundle_type":"api.community","built_at":"2022-12-20T17:06:45Z"}
{"ts":"2023-03-13T05:11:42.932Z","msg":"EnvKey API Community Edition is starting..."}
{"ts":"2023-03-13T05:11:42.934Z","msg":"Waiting for DB..."}
{"ts":"2023-03-13T05:11:48.684Z","msg":"DB migrations were unsuccessful.","alert":true,"err":{"fatal":true,"code":"PROTOCOL_CONNECTION_LOST","name":"Error","message":"Connection lost: The server closed the connection.","stack":"Error: Connection lost: The server closed the connection.\n    at Socket.<anonymous> (/tmp/webpack:/Users/daneschneider/projects/envkey/v2/node_modules/mysql2/lib/connection.js:91:31)\n    at Socket.emit (node:events:390:28)\n    at TCP.<anonymous> (node:net:687:12)"}}
Fatal error
exit status 1
```
Also, is the `DISABLE_DB_MIGRATIONS=false` need to be set on a new database ?

I tried to run in my azure cluster , the error is look like quite the same:
```
{"msg":"envkey-api binary start","api_binary_version":"2.2.3","built_from":"b1e05d63d59152522f20f353c0f80e2dedda0c89","bundle_type":"api.community","built_at":"2022-12-20T17:06:45Z"}
{"ts":"2023-03-13T05:50:30.214Z","msg":"EnvKey API Community Edition is starting..."}
{"ts":"2023-03-13T05:50:30.221Z","msg":"Waiting for DB..."}
{"ts":"2023-03-13T05:50:35.724Z","msg":"DB migrations were unsuccessful.","alert":true,"err":{"errno":-104,"code":"ECONNRESET","syscall":"read","fatal":true,"name":"Error","message":"read ECONNRESET","stack":"Error: read ECONNRESET\n    at TCP.onStreamRead (node:internal/stream_base_commons:220:20)"}}
Fatal error
exit status 1
```

Thank you in advance. Any help will be very appreciated 😄

**Comment by danenania** [comment]
Hi @trungklam it definitely looks like the problem is in establishing the DB connection. Based on the error message, it seems that the server is closing the connection for some reason. Perhaps there could be more info in the DB logs on Azure? Also, are you using MySQL 8?

**Comment by danenania** [comment]
Any luck @trungklam? Closing for now but happy to re-open if you're still stuck.

**Comment by trungklam** [comment]
Hi @danenania , sorry for late response. I was busy with other topics and can only look on this now. Yes, you are correct that the mysql 8 is the key - I was using sql server so it didnt work. So far so good my community server is up and running now.
Another question - is it possible in envkey v2 to use 1 account for multiple devices ? I noticed that for now If I logged in on another device then the current one get logged out.
Thank you a lot
